### ISSUE TITLE
Accessing non-existent property 'then' of module exports inside circular dependency

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#2095]
this only happens with a large page, and if some of the items are removed from the page, it works just fine.
The key issue is following:
Warning: Accessing non-existent property 'then' of module exports inside circular dependency
    at emitCircularRequireWarning (node:internal/modules/cjs/loader:834:11)
    at Object.get (node:internal/modules/cjs/loader:850:5)
    at <anonymous>
    at step (git/equal-access/accessibility-checker/src/lib/common/config/ACConfigManager.js:7:47)

seems there are circular dependencies among modules, and a promise-like object is used before it's fully initialized.  but don't know exactly which one.

test case: sitemap.html https://ibm-studios.slack.com/archives/C036P1CTN/p1729101710918499

**Comment by MHoov** [comment]
Assigning to @shunguoy - also, this issue is missing an estimate
